pkg/service/service.go

```diff
@@ -83,7 +84,7 @@ func NewDeviceService(serviceKey string, serviceVersion string, driver interface
 	service.driver = driver
 
 	service.config = &config.ConfigurationStruct{}
-	return &service, nil
+	return interfaces.DeviceServiceSDKExt(&service), nil
```

Comment:
we should move `Run()` to the regular DeviceService interface to make it simpler

Reply:
@cloudxxx8 , Yes. :-)

Reply:
This is fine with me. I modeled it this way based on the SecretProvider from go-mod-bootstrap, as I figured Run() would not be something we expose. If fine exposing it, then yes I can make that change.